### PR TITLE
Allow only `@DefunctConfig` class without `@Config` annotation

### PR DESCRIPTION
**Pull request by ebyhr**

### CODE REVIEW

**Review by findepi — APPROVED** [review]
Sounds reasonable

**Comment by ebyhr** [comment]
Rebased on master to resolve conflicts.

File: CHANGES [hunk 1]
```diff
@@ -1,6 +1,7 @@
 246
 
 - Exclude getter-like record methods from JSON serialization
+- Allow only `@DefunctConfig` class without `@Config` annotation
```
Comment (findepi) [inline CHANGES:4]: move to 247

Comment (ebyhr) [reply CHANGES]: Moved.

**Comment by findepi** [comment]
I should have merged this one before merging https://github.com/airlift/airlift/pull/995, apologies.
Can you rebase again, please?

**Comment by findepi** [comment]
Merged, thanks!

**Pull request merged**